small-talk/components/smolAuth.js

```diff
@@ -0,0 +1,55 @@
+import { getCookie } from "@/util/smolCookie";
+
+export function smolAuth(getServerSidePropsFunc) {
+    return async (ctx) => {
+        const { req } = ctx;
+        
+        if (req.headers.cookie) {
+            const session_id = getCookie("session_id", req.headers.cookie);
```

Comment:
Would this be replaced in the future with a unique session ID string or does the logic in getCookie ensure its unique with the regex generation anyways?

Reply:
the cookie is just a long string. it looks like this "key1=value1; key2=value2; keyn=valuen;" the regex is to perform substring operations quickly and retrieve the desired key value pair